### ISSUE TITLE
vagrant environment broken due to mismatch docker versions

### ISSUE THREAD

**Issue by csantanapr** [openwhisk/openwhisk#1409]
## Environment details:

* mac El Capitan
* vagrant 1.8.5
* virtual box 5.0.26

## Steps to reproduce the issue:

```
# Clone openwhisk
git clone --depth=1 https://github.com/openwhisk/openwhisk.git

# Change directory to tools/vagrant
cd openwhisk/tools/vagrant

# Run script to create vm and run hello action
./hello
```

## Provide the expected results and outputs:

```
wsk action invoke /whisk.system/utils/echo -p message hello --blocking --result
{
    "message": "hello"
}
```

## Provide the actual results and outputs:

```

==> default: fatal: [172.17.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Error retrieving container list: 400 Client Error: Bad Request (\"client is newer than server (client API version: 1.22, server API version: 1.21)\")"}

The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by csantanapr** [comment]
@domdom82 @markusthoemmes 
Any tips on how to fix?

Vagrant uses the scripts from ubuntu-desktop to install docker [tools/ubuntu-setup/docker.sh](https://github.com/openwhisk/openwhisk/blob/master/tools/ubuntu-setup/docker.sh)

**Comment by csantanapr** [comment]
Just checked the script it installs 1.9.1
```
sudo apt-get install -y --force-yes docker-engine=1.9.1-0~trusty
```

**Comment by markusthoemmes** [comment]
#1390 fixes this.

**Comment by csantanapr** [comment]
@markusthoemmes If the PR is not merge to make the change, why there is a mismatch?
because [tools/ubuntu-setup/ansible.sh](https://github.com/openwhisk/openwhisk/blob/master/tools/ubuntu-setup/ansible.sh#L6) was updated to install new version of docker-py to 1.9.0?

**Comment by markusthoemmes** [comment]
Yep, when i made the change to update `docker-py` I was under the impression we were already on 1.12 everywhere.

**Comment by csantanapr** [comment]
This is fixed now by #1390

**Comment by LipingSun** [comment]
Explicitly set docker API version can fix this issue: 
`echo docker_api_version=1.21 >> /etc/environment`
